### DISCUSSION TITLE
Architecture dependent livecheck

### DISCUSSION THREAD

**Discussion by pszypowicz** [Homebrew/discussions#719] [Casks]
I was trying to write an keepassxc cask formula, but it seems that the livecheck downloads wrong version for different architecture. 

Any idea how to write a proper - architecture dependent - livecheck?

https://github.com/Homebrew/homebrew-cask/pull/99397

**Comment by SMillerDev** [comment]
If there are 2 versions for 2 architectures I'd say these are just different projects.

**Reply by pszypowicz** [reply]
What if there is one version, but different patch version build, reflected in the package url?

**Reply by SMillerDev** [reply]
Will they eventually converge?
No: It's different software
Yes: You'll have to accept this inconvenience until upstream gets their version numbers in order.

**Reply by pszypowicz** [reply]
I'm in no position of experience in homebrew (I'm new to Mac OS in general), but this particular problem seems to be a product of an error, and maintainers fixing it.

I would be really suprised if such simple scenario as new arch patch build would have no simple solution in brew formulae at a cost of user inconvenience.

**Reply by SMillerDev** [reply]
> I would be really suprised if such simple scenario as new arch patch build would have no simple solution in brew formulae at a cost of user inconvenience.

This is just not a common enough case to support. It now happens sometimes for packages that are beginning to support ARM, but this is something I've never seen in 4 years as a maintainer (until a couple of months ago). For the core maintainers it simply isn't worth the effort to implement this. But you're always welcome to make a pull request if you disagree.
